feat(php): Adjust docs for Laravel DS (#5601)
getsentry/sentry-docs: src/platforms/php/guides/laravel/performance/connect-services.mdx

description: "Learn how to connect backend and frontend transactions."
---

> If you are also using Performance Monitoring for [JavaScript](/platforms/javascript/performance/) you can use helper functions to continue the trace as well as the Dynamic Sampling Context, which contains additional trace-related data that is used for trace-based sampling, started from your backend in order to [connect services](/platforms/javascript/performance/connect-services/#navigation-and-other-xhr-requests).

Author: Thanks for turning this into proper English 🙈
We do have to make clear though, that a user always needs to include both meta tags at once.

> If you are also using Performance Monitoring for [JavaScript](/platforms/javascript/performance/) you can use helper functions to continue the trace as well as the Dynamic Sampling Context, which contains additional trace-related data that is used for trace-based sampling, started from your backend in order to [connect services](/platforms/javascript/performance/connect-services/#navigation-and-other-xhr-requests).

Reviewer: Ah okay, what about this instead:
```suggestion
If you're also using Performance Monitoring for [JavaScript](/platforms/javascript/performance/), you can use helper functions to continue the trace and propagate the Dynamic Sampling context started from your backend in order to [connect services](/platforms/javascript/performance/connect-services/#navigation-and-other-xhr-requests). (The Dynamic Sampling context contains additional trace-related data that is used for trace-based sampling.)
```

Add the following lines to your blade template rendering the `<head/>` of your page:

```php {filename:app.blade.php}
<head>
...
{!! \Sentry\Laravel\Integration::sentryTracingMeta() !!}
{!! \Sentry\Laravel\Integration::sentryBaggageMeta() !!}
...
</head>
```

> These helper functions will render two meta tags similar to `<meta name="sentry-trace" content="49879feb76c84743ba5034bd2d3f1ca3-7cb5535c930d4666-1"/>` and <meta name="baggage" content="sentry-trace_id=49879feb76c84743ba5034bd2d3f1ca3,sentry-sample_rate=0.2,sentry-transaction=my-transaction,sentry-public_key=public,sentry-release=1.0.0,sentry-environment=prod>, which our JavaScript SDK will pick up and continue the trace and the Dynamic Sampling Context. Therefore, your frontend and your backend are connected by way of the same trace.

Reviewer: The second half of this sentence seems to be missing some connecting words. This part specifically doesn't make sense: "which our JavaScript SDK will pick up and continue the trace and the Dynamic Sampling context"
Should it be "will pick up and then continue the trace and generate the Dynamic Sampling context." It's not clear what elements are taking which actions here.

```suggestion
These helper functions will render two meta tags similar to `<meta name="sentry-trace" content="49879feb76c84743ba5034bd2d3f1ca3-7cb5535c930d4666-1"/>` and `<meta name="baggage" content="sentry-trace_id=49879feb76c84743ba5034bd2d3f1ca3,sentry-sample_rate=0.2,sentry-transaction=my-transaction,sentry-public_key=public,sentry-release=1.0.0,sentry-environment=prod>`, which our JavaScript SDK will pick up and continue the trace and the Dynamic Sampling context. Therefore, your frontend and your backend are connected by way of the same trace.
```

Author: `sentry-trace` continues the trace, `sentry-baggage` will be used to propagate the Dynamic Sampling context.

Reviewer: Okay what about this:
```suggestion
These helper functions will render two meta tags similar to `<meta name="sentry-trace" content="49879feb76c84743ba5034bd2d3f1ca3-7cb5535c930d4666-1"/>` and `<meta name="baggage" content="sentry-trace_id=49879feb76c84743ba5034bd2d3f1ca3,sentry-sample_rate=0.2,sentry-transaction=my-transaction,sentry-public_key=public,sentry-release=1.0.0,sentry-environment=prod>`. Our JavaScript SDK will pick these up and then continue the trace and propagate the Dynamic Sampling context. 
As a result, your frontend and your backend are connected by way of the same trace.
```
